datastore/datastore.go

```diff
@@ -253,6 +254,32 @@ func NewDataStore(scope string, cfg *ScopeCfg) (DataStore, error) {
 	return newClient(scope, cfg.Client.Provider, cfg.Client.Address, cfg.Client.Config, cached)
 }
 
+// NewDataStoreFromConfig creates a new instance of LibKV data store starting from the datastore config data
+func NewDataStoreFromConfig(dsc discoverapi.DatastoreConfigData) (DataStore, error) {
+	if dsc.Config == nil {
+		return nil, fmt.Errorf("store configuration has nil store config")
```

Comment:
Actually the nil check is not needed, dsc.Config is an interface.
The casting to `*store.Config` will work as expected, it will be nil.